sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/_client.py

```diff
@@ -3,266 +3,717 @@
 # Licensed under the MIT License. See LICENSE.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-import functools
-from typing import Any, List, Mapping, Optional
-import uuid
-
-from ._models import (
-    DeletedKey,
-    DeletedKeyItem,
-    DeletedKeyItemPaged,
-    JsonWebKey,
-    Key,
-    KeyAttributes,
-    KeyCreateParameters,
-    KeyItem,
-    KeyItemPaged,
-    KeyUpdateParameters,
-)
-from azure.core.configuration import Configuration
-from azure.core.exceptions import HttpResponseError
+from typing import Any, Dict, Generator, Mapping, Optional, List
+from datetime import datetime
+
+from azure.core import Configuration
+from azure.core.exceptions import ResourceExistsError, ResourceNotFoundError
 from azure.core.pipeline import Pipeline
-from azure.core.pipeline.policies import (
-    HTTPPolicy,
-    UserAgentPolicy,
-    HeadersPolicy,
-    RetryPolicy,
-    RedirectPolicy,
-    ContentDecodePolicy,
-)
-from azure.core.pipeline.transport import RequestsTransport, HttpRequest, HttpResponse, HttpTransport
-from msrest import Serializer, Deserializer
+from azure.core.pipeline.transport import RequestsTransport
+from azure.security.keyvault._internal import _BearerTokenCredentialPolicy
 
+from .._generated import KeyVaultClient
+from ._models import Key, KeyBase, DeletedKey
+import collections
 
-class BearerTokenCredentialPolicy(HTTPPolicy):
-    def __init__(self, credentials):
-        self._credentials = credentials
 
-    def send(self, request, **kwargs):
-        # type: (HttpRequest, Any) -> HttpResponse
-        auth_header = "Bearer " + self._credentials.token["access_token"]
-        request.http_request.headers["Authorization"] = auth_header
+class KeyClient:
+    """KeyClient defines a high level interface for
+    managing keys in the specified vault.
 
-        return self.next.send(request, **kwargs)
+    :param credentials:  A credential or credential provider which can be used to authenticate to the vault,
+        a ValueError will be raised if the entity is not provided
+    :type credentials: azure.authentication.Credential or azure.authentication.CredentialProvider
+    :param str vault_url: The url of the vault to which the client will connect,
+        a ValueError will be raised if the entity is not provided
+    :param ~azure.core.configuration.Configuration config:  The configuration for the KeyClient
 
+    Example:
+        .. literalinclude:: ../tests/test_examples_keyvault.py
+            :start-after: [START create_key_client]
+            :end-before: [END create_key_client]
+            :language: python
+            :dedent: 4
+            :caption: Creates a new instance of the Key client
+    """
 
-class KeyClient:
-    API_VERSION = "7.0"
+    KeyOperationResult = collections.namedtuple("KeyOperationResult", "id value")
```

Comment:
The second argument should be a list; see [`_VaultId`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/keyvault/azure-security-keyvault/azure/security/keyvault/_internal.py#L16) for an example. Is there any particular reason to define this on `KeyClient` rather than with the models?